### PR TITLE
Translation and corrections for Hebrew #1582

### PR DESCRIPTION
**Pull request by samuelgozi**
Translated whatever I could and also reviewed some of the already translated lines.
I didn't build the GUI, but Im confident that my translations are good.

### CODE REVIEW

File: translations/monero-core_he.ts [hunk 1]
```diff
@@ -1168,38 +1175,38 @@ The old wallet cache file will be renamed and can be restored later.
...
         <location filename="../pages/settings/SettingsNode.qml" line="323"/>
         <source>Password</source>
-        <translation type="unfinished">סיסמה</translation>
+        <translation>סיסמה</translation>
     </message>
     <message>
         <location filename="../pages/settings/SettingsNode.qml" line="347"/>
         <source>Connect</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">התחבר</translation>
```
Comment (erciccione) [inline translations/monero-core_he.ts]: If the translation is correct, you can remove `type="unfinished"` here

**Comment by samuelgozi** [comment]
Did my best to fix what you pointed out as well as adding some other small changes and translations.

**Comment by samuelgozi** [comment]
@erciccione Fixed that.

**Comment by samuelgozi** [comment]
I added you as a collaborator

**Pull request merged**